app/policies/page_policy.rb

```diff
@@ -0,0 +1,14 @@
+class PagePolicy < ApplicationPolicy
+
+  def destroy?
+    @user.role == "administrator" && @record.user_id == @user.id
```

Comment:
I suggest to create method 'administrator?' inside user model and rewrite this like

@user.administrator? && @record.user == @user